packages/nimble-components/src/table/specs/table-column-specs/table-column-formatted-text.md

```diff
@@ -161,21 +161,65 @@ Follows the [Column Type Philosophy](/packages/nimble-components/src/table/specs
+
+When displaying units, `Intl.NumberFormat` will translate unit strings for the [units that it supports](https://tc39.es/ecma402/#table-sanctioned-single-unit-identifiers) and localize the number (for comma/decimal). We will include our own logic for converting between unit values. For a family of units not supported by `Intl.NumberFormat`, we will provide our own translations (for some subset of languages) in a `nimble-unit-<name>` element. If the client requests a translation for one of these units in a language we don't support, we will fall back to English.
+
+Unit elements will be capable of enumerating the individual units supported. The enumerated unit objects will be capable of formatting a given number into a localized string including the unit label. Below is an example of the abstractions/APIs that might be used to implement this.
+
+```TS
+// a specific unit, e.g. kilobyte, millivolt, etc.
+interface Unit {
+    public conversionFactor: number;
+    public format(value: number): string;
+}
+
+// a set of related units, e.g. {byte, kilobyte, megabyte, gigabyte, terabyte, petabyte}
+interface UnitFamily {
+    public getSupportedUnits(lang: string, formatterOptions: Intl.NumberFormatOptions): Unit[];
+}
+
+// unit family supported by Intl.NumberFormat
+class UnitFamilyByte extends FoundationElement implements UnitFamily {
+    public getSupportedUnits(lang: string, formatterOptions: Intl.NumberFormatOptions): Unit[] {
+        // returns implementations of Unit that wrap Intl.NumberFormat instances configured for a specific locale and a specific unit (e.g. 'fr-FR' and 'kilobyte')
+    }
+}
+
+// unit family with Nimble-provided unit translations
+class UnitFamilyVolt extends FoundationElement implements UnitFamily {
+    public getSupportedUnits(lang: string, formatterOptions: Intl.NumberFormatOptions): Unit[] {
+        // returns implementations of Unit that contain a shared Intl.NumberFormat (for formatting the number) and a specific translated unit string to append
+    }
+}
+```
+
+We will use `nimble-unit-byte` to display file sizes in SLE tables. Currently, SLE displays these values with the common `KB`/`MB`/`GB` unit labels, but uses a factor of 1024 to convert between units (which is [not uncommon](https://en.wikipedia.org/wiki/JEDEC_memory_standards#Unit_prefixes_for_semiconductor_storage_capacity), but [technically incorrect](https://physics.nist.gov/cuu/Units/binary.html)). Our formatting will use the same unit labels, but use a conversion factor of 1000. This will result in slightly different values being displayed. For SLE's internal consistency, we will also change their file size pipe to use a conversion factor of 1000 instead of 1024. We will need to search for other places in the product where byte values are being converted and update them similarly.
```

Comment:
To help clients maintain consistent formatting outside of the table, in the duration column we ended up deciding to expose a TS class with the formatting logic and an Angular pipe that wraps it. I think in your previous HLD you mentioned that might be low ROI but I'd like to revisit that in a quest for consistency with the duration column.

The challenge here is that there is potentially more configuration to specify when formatting a unit value. How would a client specify the unit family and any additional configuration to the format?

I don't have great ideas at the moment. If others don't either then maybe we can defer this until we start uptaking this in SLE.

Reply:
Essentially duplicate of https://github.com/ni/nimble/pull/1596#discussion_r1364236902